### PR TITLE
added cancel button

### PR DESCRIPTION
**Pull request by anamika7153**
Closes: #756 

### Video

https://github.com/narayan954/dummygram/assets/66116440/ef642e16-c73d-4bee-9da4-c93f22edc17d

### CODE REVIEW

File: src/pages/Profile/index.jsx [hunk 1]
```diff
@@ -235,8 +242,9 @@ function Profile() {
             });
           })
           .catch((error) => console.error(error));
-      },
+      }
     );
+    setProfilePic(null);
```
Comment (narayan954) [inline src/pages/Profile/index.jsx:247]: again same doubt

**Review by narayan954 — CHANGES_REQUESTED** [review]
some minor changes requested. Also a doubt, are you replacing visible with profilepic?

**Comment by anamika7153** [comment]
> some minor changes requested. Also a doubt, are you replacing visible with profilepic?

Yeah, when I added the cancel btn and cancel function visible state wasn't changing from true to false even after updating the state. Issue might be with the order of state updates. I couldn't solve it. udating the profilepic worked. So I tested it and used it.

File: src/pages/Profile/index.jsx [hunk 2]
```diff
@@ -208,6 +209,12 @@ function Profile() {
     }
   };
 
+  const handleCancel = () => {
+    setProfilePic(null);
```
[the remark this answers is not in the document]
Comment (anamika7153) [reply src/pages/Profile/index.jsx:213]: > should we set it to null or the actual avatar?

I needed to render the buttons conditionally. That is why I set it to null. Also, Setting it to null wasn't creating any issues with the original avatar.

**Comment by narayan954** [comment]
> > some minor changes requested. Also a doubt, are you replacing visible with profilepic?
> 
> Yeah, when I added the cancel btn and cancel function visible state wasn't changing from true to false even after updating the state. Issue might be with the order of state updates. I couldn't solve it. udating the profilepic worked. So I tested it and used it.

let's merge the pic updating issue first as it can cause conflicts

**Comment by anamika7153** [comment]
> > > some minor changes requested. Also a doubt, are you replacing visible with profilepic?
> > 
> > 
> > Yeah, when I added the cancel btn and cancel function visible state wasn't changing from true to false even after updating the state. Issue might be with the order of state updates. I couldn't solve it. udating the profilepic worked. So I tested it and used it.
> 
> let's merge the pr for pic updating issue first as it can cause conflicts with this pr. I've commented there

Okay, I'll notify when I'm done with that.

File: src/pages/Profile/index.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (narayan954) [reply src/pages/Profile/index.jsx:213]: Ohh, I think variable names are creating confusions, that's why I had this doubt

**Comment by narayan954** [comment]
closing as duplicate to #926

**Pull request closed without merging**